umm-lib/src/cmr/umm/iso_smap/granule/spatial.clj

```diff
@@ -1,20 +1,30 @@
 (ns cmr.umm.iso-smap.granule.spatial
   "Contains functions for parsing and generating the ISO SMAP granule spatial"
-  (:require [cmr.common.xml :as cx]
-            [cmr.umm.umm-granule :as g]
-            [cmr.umm.iso-smap.spatial :as spatial]))
+  (:require
+   [cmr.common.xml :as cx]
+   [cmr.common.util :as util]
+   [cmr.umm.umm-granule :as g]
+   [cmr.umm.iso-smap.spatial :as spatial]))
 
 (def extent-path
   [:composedOf :DS_DataSet :has :MI_Metadata :identificationInfo :MD_DataIdentification
    :extent :EX_Extent :geographicElement])
 
+(defn- orbit?
+  "Returns true or false depending on the existence of the :ascending-crossing key existing in the
+  spatial coverage map."
+  [spatial-map]
+  (some #(= :ascending-crossing %) (keys spatial-map)))
+
 (defn xml-elem->SpatialCoverage
   "Returns a UMM SpatialCoverage from a parsed XML structure"
   [xml-struct]
   (let [spatial-elems (cx/elements-at-path xml-struct extent-path)
         geometries (flatten (map spatial/decode spatial-elems))]
     (when (seq geometries)
-      (g/map->SpatialCoverage {:geometries geometries}))))
+      (g/map->SpatialCoverage (util/remove-map-keys empty?
+                                {:geometries (remove #(orbit? %) geometries)
```

Comment:
When a function only takes one argument you can write it as `(remove orbit? geometries)`. Similar for the `filter` on the next line.